### PR TITLE
Add io.github.garglk.Gargoyle

### PR DESCRIPTION
**Pull request by cspiegel**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

There are two permissions which probably require explanation:

1. `--filesystem=home`
    * While Gargoyle, via Qt, can use portals to open files in general, there are sometimes auxiliary files which need to be loaded simultaneously, but which are not specified by the user; instead, they're searched for in the same directory as the "main" file, and as such, C and C++ I/O is directly used to open and read them.
2. `--socket=system-bus`
    * Without this, the file picker on Plasma dumps the following to stderr:
    ```
    kf.solid.backends.udisks2: Failed enumerating UDisks2 objects: "org.freedesktop.DBus.Error.Disconnected"
     "Not connected to D-Bus server"
    ```
    Despite this, as far as I can tell, all disks are still visible in the file chooser, so I'm not entirely sure this is anything but a red herring.  But I'd rather get feedback on this before removing it, in case it's something that's been seen before.

A quick note on the file `fdo.patch`: this fixes a few things up to conform to freedesktop.org standards. This is currently in a branch upstream that I'll merge after Gargoyle goes through the Flatpak process, so that the two are in sync.

### CODE REVIEW

File: io.github.garglk.Gargoyle.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+command: gargoyle
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --device=dri
+  - --socket=system-bus
```
Comment (hfiguiere) [inline io.github.garglk.Gargoyle.yaml:14]: no. you should figure out which d-bus name it needs. use `--log-system-bus` with `flatpak run` to figure out what is needed.

File: io.github.garglk.Gargoyle.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+id: io.github.garglk.Gargoyle
+runtime: org.kde.Platform
+runtime_version: "5.15-21.08"
+sdk: org.kde.Sdk
+command: gargoyle
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline io.github.garglk.Gargoyle.yaml:10]:
```suggestion
  - --socket=fallback-x11
```

File: fdo.patch [hunk 3]
```diff
@@ -0,0 +1,92 @@
+diff --git a/garglk/CMakeLists.txt b/garglk/CMakeLists.txt
+index 2d9b935e..4dd4167d 100644
+--- a/garglk/CMakeLists.txt
++++ b/garglk/CMakeLists.txt
+@@ -239,9 +239,9 @@ elseif(UNIX)
+     endif()
+ 
+     if(WITH_FREEDESKTOP)
+-        install(FILES "gargoyle.desktop" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications")
+-        install(FILES "gargoyle-house.png" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons")
```
Comment (hfiguiere) [inline fdo.patch:10]: I would just make it install it in the right place

File: fdo.patch [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
++    <launchable type="desktop-id">io.github.garglk.Gargoyle.desktop</launchable>
++    <releases>
++        <release version="2022.1" date="2022-01-30"/>
++        <release version="2019.1.1" date="2019-11-19"/>
++        <release version="2019.1" date="2019-11-03"/>
++        <release version="2011.1" date="2011-10-19"/>
++    </releases>
++    <provides>
++        <binary>gargoyle</binary>
++    </provides>
```
Comment (hfiguiere) [inline fdo.patch:90]: the appstream file is missing `content_rating`.

File: fdo.patch [hunk 5]
```diff
@@ -0,0 +1,92 @@
+diff --git a/garglk/CMakeLists.txt b/garglk/CMakeLists.txt
```
Comment (hfiguiere) [inline fdo.patch:1]: has this patch been submitted upstream?

File: io.github.garglk.Gargoyle.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --device=dri
+  - --socket=system-bus
+  - --filesystem=home
+modules:
+  - name: gargoyle
+    buildsystem: cmake
```
Comment (hfiguiere) [inline io.github.garglk.Gargoyle.yaml:18]:
```suggestion
    buildsystem: cmake-ninja
```

this is usually faster.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build io.github.garglk.Gargoyle

File: io.github.garglk.Gargoyle.yaml (hunk 1 above)
Comment (cspiegel) [reply io.github.garglk.Gargoyle.yaml:14]: As mentioned, this was to get feedback on an error message.  There's nothing that's output with `--log-system-bus`, even though the error message only disappears when there's a system bus (session bus isn't enough, though there _is_ session bus logging).  As in, providing access to the system bus is the only way to remove the error, but there's apparently nothing accessing the system bus, according to the logging?  Not sure what's happening here, and it's possible it's some idiosyncratic behavior on my system.

Again, this doesn't _seem_ to affect behavior, but I'd obviously like to determine why the error message is there.

File: fdo.patch (hunk 4 above)
Comment (hfiguiere) [reply fdo.patch:90]: As I said:

```
builddir/files/share/appdata/io.github.garglk.Gargoyle.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
Validation of files failed
```

File: fdo.patch (hunk 3 above)
Comment (cspiegel) [reply fdo.patch:10]: I'm not entirely sure what you mean.  This is a low-resolution PNG that doesn't satisfy the requirements for Flathub, so it's not installed here, and instead a replacement SVG is being installed.  At some point we'll switch the official icon to an SVG, but the current release just doesn't have one.

File: fdo.patch (hunk 5 above)
Comment (cspiegel) [reply fdo.patch:1]: As I mentioned in the pull request, this comes from an upstream branch that will be merged after it's synchronized with any changes required for Flathub.

**Comment by hfiguiere** [comment]
bot, build io.github.garglk.Gargoyle

**Comment by hfiguiere** [comment]
bot, build io.github.garglk.Gargoyle

File: io.github.garglk.Gargoyle.yaml (hunk 1 above)
Comment (Erick555) [reply io.github.garglk.Gargoyle.yaml]: Could you try `--system-talk-name=org.freedesktop.UDisks2`?

Comment (cspiegel) [reply io.github.garglk.Gargoyle.yaml]: That seems to do the trick, thanks!

File: io.github.garglk.Gargoyle.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  - --socket=pulseaudio
+  - --device=dri
+  - --filesystem=home
+  - --system-talk-name=org.freedesktop.UDisks2
+modules:
+  - name: gargoyle
+    buildsystem: cmake-ninja
+    builddir: true
+    post-install:
+      - install -D $FLATPAK_BUILDER_BUILDDIR/gargoyle-house.svg /app/share/icons/hicolor/scalable/apps/io.github.garglk.Gargoyle.svg
```
Comment (hfiguiere) [inline io.github.garglk.Gargoyle.yaml:20]:
```suggestion
      - install -D ../gargoyle-house.svg /app/share/icons/hicolor/scalable/apps/io.github.garglk.Gargoyle.svg
```

Ideally this icon would be upstream

Comment (cspiegel) [reply io.github.garglk.Gargoyle.yaml:20]: At the moment we have a low-res icon upstream.  I intend to include an SVG icon in the next release, but it's not yet been determined what that icon will be.  The icon here is an SVG representation of the current icon (a white house), so fits with the appearance of the current release, but may not be what ends up as the final icon.  When the next release comes out, this icon will be removed from the Flathub repo, and either added to the official release (if it's the one being used), or the release will have another SVG icon that will be suitable for Flathub.

**Comment by barthalion** [comment]
bot, build io.github.garglk.Gargoyle

**Pull request closed without merging**